okhttp/src/main/java/okhttp3/internal/Util.kt

```diff
@@ -0,0 +1,663 @@
+import okhttp3.internal.http2.Header
+import okhttp3.internal.platform.Platform
+import okio.Buffer
+import okio.BufferedSource
+import okio.ByteString.Companion.decodeHex
+import okio.Options
+import okio.Source
+import java.io.Closeable
+import java.io.IOException
+import java.io.InterruptedIOException
+import java.net.IDN
+import java.net.InetAddress
+import java.net.ServerSocket
+import java.net.Socket
+import java.net.UnknownHostException
+import java.nio.charset.Charset
+import java.nio.charset.StandardCharsets.UTF_16BE
+import java.nio.charset.StandardCharsets.UTF_16LE
+import java.nio.charset.StandardCharsets.UTF_8
+import java.security.AccessControlException
+import java.util.ArrayList
+import java.util.Arrays
+import java.util.Arrays.asList
+import java.util.Collections
+import java.util.Comparator
+import java.util.LinkedHashMap
+import java.util.Locale
+import java.util.TimeZone
+import java.util.concurrent.ThreadFactory
+import java.util.concurrent.TimeUnit
+import java.util.regex.Pattern
+import javax.net.ssl.X509TrustManager
+
+/** Junk drawer of utility methods.  */
+object Util {
+  @JvmField
+  val EMPTY_BYTE_ARRAY = ByteArray(0)
+  @JvmField
+  val EMPTY_HEADERS = Headers.of()
+
+  @JvmField
+  val EMPTY_RESPONSE = ResponseBody.create(null, EMPTY_BYTE_ARRAY)
+  @JvmField
+  val EMPTY_REQUEST = RequestBody.create(null, EMPTY_BYTE_ARRAY)
+
+  /** Byte order marks.  */
+  private val UNICODE_BOMS = Options.of(
+      "efbbbf".decodeHex(), // UTF-8
+      "feff".decodeHex(), // UTF-16BE
+      "fffe".decodeHex(), // UTF-16LE
+      "0000ffff".decodeHex(), // UTF-32BE
+      "ffff0000".decodeHex() // UTF-32LE
+  )
+
+  private val UTF_32BE = Charset.forName("UTF-32BE")
+  private val UTF_32LE = Charset.forName("UTF-32LE")
+
+  /** GMT and UTC are equivalent for our purposes.  */
+  @JvmField val UTC = TimeZone.getTimeZone("GMT")
+
+  val NATURAL_ORDER = Comparator<String> { obj, anotherString -> obj.compareTo(anotherString) }
+
+  /**
+   * Quick and dirty pattern to differentiate IP addresses from hostnames. This is an approximation
+   * of Android's private InetAddress#isNumeric API.
+   *
+   *
+   * This matches IPv6 addresses as a hex string containing at least one colon, and possibly
+   * including dots after the first colon. It matches IPv4 addresses as strings containing only
+   * decimal digits and dots. This pattern matches strings like "a:.23" and "54" that are neither IP
+   * addresses nor hostnames; they will be verified as IP addresses (which is a more strict
+   * verification).
+   */
+  private val VERIFY_AS_IP_ADDRESS = Pattern.compile(
+      "([0-9a-fA-F]*:[0-9a-fA-F:.]*)|([\\d.]+)")
+
+  fun checkOffsetAndCount(arrayLength: Long, offset: Long, count: Long) {
+    if (offset or count < 0 || offset > arrayLength || arrayLength - offset < count) {
+      throw ArrayIndexOutOfBoundsException()
+    }
+  }
+
+  /**
+   * Closes `closeable`, ignoring any checked exceptions. Does nothing if `closeable` is
```

Comment:
```suggestion
   * Closes [closeable], ignoring any checked exceptions. Does nothing if [closeable] is
```

Reply:
There's a _lot_ of these documentation improvements like this for this file. I can take it once this is merged if you want.

Reply:
I think I've got them covered, thanks for flagging.